lib/household/view/view_household.dart

```diff
@@ -0,0 +1,157 @@
+          ),
+        );
+      },
+    );
+  }
+}
+
+Widget householdView(context, state) {
+  if (state is HouseholdInitial) {
+    return const Center(child: CircularProgressIndicator());
+  } else if (state is HouseholdLoaded) {
+    DateFormat formatter = DateFormat('yyyy-MM-dd');
+
+    return Column(
+      children: [
+        TextFormField(
+          readOnly: true,
+          decoration: const InputDecoration(
+              labelText: 'Location', icon: Icon(Icons.location_on_outlined)),
+          initialValue: state.household!.geoLocation,
+        ),
+        TextFormField(
+          readOnly: true,
+          decoration: const InputDecoration(
+              labelText: 'Sensitization Date',
+              icon: Icon(Icons.calendar_today)),
+          initialValue: formatter.format(state.household!.sensitizationDate),
+        ),
+        TextFormField(
+          readOnly: true,
+          decoration: const InputDecoration(
+              labelText: 'Comments', icon: Icon(Icons.comment)),
+          initialValue: state.household!.comments,
+          minLines: 1,
+          maxLines: null,
+        ),
+        const Padding(
+          padding: EdgeInsets.all(8.0),
+          child: Divider(),
+        ),
+        Row(
+          children: [
+            const Padding(
+              padding: EdgeInsets.all(8.0),
+              child: Icon(Icons.people),
+            ),
+            Text('Respondents', style: Theme.of(context).textTheme.headline6),
+            const Spacer(),
+            IconButton(
+                onPressed: () =>
+                    Navigator.pushNamed(context, PageRouter.createRespondent),
+                icon: const Icon(Icons.add))
+          ],
+        ),
+        Expanded(
+          child: ListView.builder(
+            padding: const EdgeInsets.all(2.0),
+            itemCount: state.household!.respondents.length,
+            itemBuilder: (context, index) {
+              return Card(
+                  child: ListTile(
+                title: Text(state.household!.respondents[index].name),
+                subtitle: Text("ID: ${index + 1}"),
+                onTap: () => {
+                  context
+                      .read<HouseholdBloc>()
+                      .add(RespondentOpened(index: index)),
+                  Navigator.pushNamed(context, PageRouter.viewRespondent,
+                      arguments: {'index': index})
+                },
+                onLongPress: () => showModalBottomSheet(
+                    context: context,
+                    builder: (context) {
+                      return RespondentOptions(
+                          index: index,
+                          name: state.household!.respondents[index].name);
+                    }),
+              ));
+            },
+          ),
+        )
+      ],
+    );
+  } else {
+    return const Center(child: Text('Error'));
+  }
+}
+
+class RespondentOptions extends StatelessWidget {
```

Comment:
same with this one

Reply:
Since this is constructing a bottom sheet rather than a widget inside the existing scaffold I'd prefer to keep it separate

Reply:
Done